apps/examples/smartfs_powercut/smartfs_powercut_main.c

```diff
@@ -171,98 +212,151 @@ static int create_file(char *path, ssize_t bufsize, file_type_t type)
 {
 	int fd;
+	int fd_sample = 0;
 	int ret;
-	ssize_t len;
 	off_t size;
 	int i;
-	uint16_t crc;
 	struct stat st;
-	int data_len = bufsize - sizeof(g_buf.crc);
+	uint16_t crc1;
+	uint16_t crc2;
+	uint16_t data_len;
 	
 	fd = open(path, O_RDWR);
 	if (fd < 0) {
 		if (errno == ENOENT) {
-			printf("Verify_file File is not exist, no need to verify!! path : %s\n", path);
+			printf("File to be verified does not exist, no need to verify!! path : %s\n", path);
 			*index = FILE_CHUNK_NO_INDEX;
 			return OK;
 		}
 		
-		printf("verify_file %s open error... errno : %d\n", path, errno);
+		printf("Unable to open file : %s for verifying, errno : %d\n", path, errno);
 		return ERROR;
 	}
 
+	data_len = bufsize - sizeof(g_buf.crc);
+
+	if (type == FILE_TYPE_TEXT) {
+		fd_sample = open(SAMPLE_TXT_FILE, O_RDOK);
+		if (fd_sample < 0) {
+			printf("Unable to open Sample Text File : %s, errno : %d\n", SAMPLE_TXT_FILE, errno);
+			ret = ERROR;
+			goto error_with_fd;
+		}
+	}
+
 	for (i = 0; i < TEST_WRITECOUNT; i++) {
-		len = read(fd, (void *)&g_buf, bufsize);
-		if (len != bufsize) {
-			/* If write is not finished properly */
-			printf("verify_file datachunk is not written fully..  path : %s ret : %d index : %d\n", path, len, i);
-			break;
-		} else {
-			/* Maybe this won't be happened but check crc for double check */
-			crc = crc16((const uint8_t *)g_buf.data, data_len);
-			if (crc != g_buf.crc) {
-				printf("verify_file crc mismatch!! path : %s calulcated crc : %d written crc : %d index : %d\n", path, crc, g_buf.crc, i);
+		if (type == FILE_TYPE_TEXT) {
+			ret = read(fd, (void *)g_buf.data, bufsize);
+			if (ret != bufsize) {
+				printf("Unable to read file: %s, ret: %d\n", path, ret);
 				break;
 			}
+			crc1 = crc16((const uint8_t *)g_buf.data, bufsize);
+
+			ret = read(fd_sample, (void *)g_buf.data, bufsize);
+			if (ret != bufsize) {
+				printf("Failed to read sample file: %s, ret: %d\n", path, ret);
+				ret = ERROR;
+				goto error_with_fd_sample;
+			}
+			crc2 = crc16((const uint8_t *)g_buf.data, bufsize);
+		}  else {
+			ret = read(fd, (void *)&g_buf, bufsize);
+			if (ret != bufsize) {
+				printf("Unable to read file: %s, ret: %d\n", path, ret);
+				break;
+			}
+			crc1 = crc16((const uint8_t *)g_buf.data, data_len);
+			crc2 = g_buf.crc;
+		}
+
+		if (crc1 != crc2) {
+			printf("Verify file : %s failed at index %d, crc1 : %d, crc2 : %d\n", path, i, crc1, crc2);
+			break;
 		}
 	}
 
+	close(fd_sample);
 	*index = i;
-
-	/* shrink to last valid index here, it is necessary to bury invalid data chunk */
+	/* Shrink file to last valid index here, it is necessary to bury invalid data chunk */
 	if ((i > 0) && (i < TEST_WRITECOUNT)) {
 		/* truncate current file to include valid data only */
 		size = i * bufsize;
 		
 		ret = lseek(fd, 0, SEEK_SET);
 		if (ret != OK) {
-			printf("verify_file, seek to 0 failed\n");
+			printf("Verify_file : %s, seek to 0 failed, ret : %d, errno : %d\n", path, ret, errno);
```

Comment:
oops my mistake. lseek returns newpos so if (ret != 0) is correct. (it works now because OK is 0 but it is wrong)

Reply:
Updated